### PR TITLE
Circular Queue data structure in C-Plus-Plus

### PR DESCRIPTION
**Pull request by anuragmukherjee2001**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes, I have read the contribution guidelines.

### Description

I am participating in GSSoC'21, so I want to contribute to this project. Apart from this, I found this repository very knowledgeable, consisting data structures and algorithms of different programming languages.

I contributed a code on circular queue data structure.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

I created an issue. The link of the issue is:
#2768

### CODE REVIEW

**Comment by anuragmukherjee2001** [comment]
Please review this Pull Request.

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Few changes:
- Use multi-line comments
- Add sample I/O at the end of the Code
- Add complexity at the end of the Code
- Add a description of Code at the top
- Add a new line at the EOF

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Use multi-line comments

File: C-Plus-Plus/README.md [hunk 1]
```diff
@@ -92,9 +92,13 @@
 - [Binary Search Tree to Greater Sum Tree](ds/BST_to_Greater_Sum_Tree.cpp)
 - [Threaded Binary Tree](ds/ThreadedBinaryTree.cpp)
 - [Alternate Merging of nodes of two linked lists](ds/alternate_merging_of_linked_list.cpp)
+
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Remove extra line

File: C-Plus-Plus/README.md [hunk 2]
```diff
@@ -92,9 +92,13 @@
...
+- [Circular Queue](ds/Circular_Queue.cpp)
+
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Remove extra line

File: C-Plus-Plus/ds/Circular_Queue.cpp [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+}
+
+/*
+ The time complexities of different operations in circular queue data structure is:-
+
+  1) The time complexity of enqueue operation is O(1), that is constant time.
+  2) The time complexity of dequeue operation is O(1), that is constant time. 
+  3) The time complexity for checking whether the queue is empty of full is O(1), that is constant time.
+  4) The time complexity of display function is O(n), where n is the number of Elements in the queue.
+*/
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/Circular_Queue.cpp:258]: Add sample I/O

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by anuragmukherjee2001** [comment]
A sample input and output is added and all the extra lines are removed, as per the told instructions. Please check it sir.

File: C-Plus-Plus/ds/Circular_Queue.cpp [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+  2. Delete element from the queue - Dequeue
+  3. Display items in the queue
+  4. To check whether the queue is empty or not
+  5. To check whether the queue is full or not
+  6. Exit
+  Enter your choice
+  6 
+*/
+/*
+ The time complexities of different operations in circular queue data structure is:-
```
Comment (atarax665) [inline C-Plus-Plus/ds/Circular_Queue.cpp]:
```suggestion
 The time complexities of different operations in a circular queue data structure are:-
```

File: C-Plus-Plus/ds/Circular_Queue.cpp [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+
+#define max 5
+
+using namespace std;
+
+// Writing a class where all the basic functions of the circular queue presides
+
+class CircularQueue{
+    /*
+    Defining all the variables as public variables.
```
Comment (atarax665) [inline C-Plus-Plus/ds/Circular_Queue.cpp:22]: Inconsistent indentation.

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/ds/Circular_Queue.cpp (hunk 5 above)
Comment (anuragmukherjee2001) [reply C-Plus-Plus/ds/Circular_Queue.cpp:22]: Sir I corrected the indentation and suggested changes. Please check it sir.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**